battle_move_test.go

```diff
@@ -19,6 +19,78 @@ func registerMoveWithType(t Type) MoveId {
 
 var TestMoveDefault = RegisterMove(MoveData{Name: "Default", Category: MoveCategoryPhysical, Power: 10, Accuracy: 100, InitialMaxPP: 100})
 var TestMoveNoDamage = RegisterMove(MoveData{Name: "No Damage", InitialMaxPP: 100})
+var TestMoveInflictBurn = RegisterMove(MoveData{Name: "Inflict Burn", Category: MoveCategoryStatus, InitialMaxPP: 100, Ailment: StatusBurn})
+var TestMoveDamageAndInflictBurn = RegisterMove(MoveData{Name: "Damage and Inflict Burn", Category: MoveCategoryPhysical, InitialMaxPP: 100, AilmentChance: 100, Ailment: StatusBurn, Power: 10})
+
+var _ = Describe("Move Status Inflict", func() {
+	a := Agent(new(dumbAgent))
+	It("should inflict burn", func() {
+		b := New1v1Battle(
+			GeneratePokemon(PkmnSquirtle, WithMoves(TestMoveInflictBurn)), &a,
+			GeneratePokemon(PkmnSquirtle, WithMoves(TestMoveNoDamage)), &a,
+		)
+		b.rng = AlwaysRNG()
```

Comment:
~Not sure if any RNG is necessary here, because moves with 0 accuracy will not miss.~
Forgot that status effects are RNG...